docs/requirements.md

```diff
@@ -2,11 +2,23 @@
 
 Running Codacy on a Kubernetes cluster requires the following:
 
--   A Kubernetes 1.14.\* or 1.15.\* cluster provisioned with the [required resources](#cluster-resource-requirements)
+-   A Kubernetes 1.14.\* or 1.15.\* cluster provisioned with the [required resources](#hardware-requirements)
 -   The [NGINX Ingress Controller](https://github.com/helm/charts/tree/master/stable/nginx-ingress) for Kubernetes
 -   A [PostgreSQL server](#postgresql-server-setup) accessible from the Kubernetes cluster
 
-## Cluster resource requirements
+## Hardware Requirements
+
+The resource usage for Codacy depends a lot on the amount of concurrent analysis that you configure. Check the following table for some examples.
+
+| Installation type              | Max. parallel analysis | vCPUs | Memory | Storage |
+| ------------------------------ | ---------------------- | ----- | ------ | ------- |
+| Kubernetes Small Installation  | 2                      | 16    | 64 GB  | 1 TB+   |
+| Kubernetes Medium Installation | 4                      | 32    | 128 GB | 1 TB+   |
+| Kubernetes Big Installation    | 10                     | 64    | 256 GB | 1 TB+   |
+| MicroK8s Minimum               | 2                      | 16    | 32 GB  | 1 TB+   |
+| MicroK8s Recommended           | 2                      | 16    | 64 GB  | 1 TB+   |
```

Comment:
@paulopontesm this new table with examples is really helpful!

I have two questions:
* Depending on the type / size of installation, the user must adjust the values in the reference `values-[production|microk8s].yaml` files, correct? If so, this mapping should be made clearer below when we mention the reference `.yaml` files.
* What are the differences between a small / medium / big installation? Does it only have to do with the number of analysis running in parallel? Or are there any other factors that affect this, such as number of users simultaneously using the website or the number and size of the repositories?

Besides this, right now the MicroK8s and Kubernetes hardware requirements are not "parallel", since for MicroK8s we have "minimum and recommended" while for Kubernetes we have "small / medium / big" installations. This could cause confusion when trying to directly compare the two types of cluster. For example, as a customer, I would think that the MicroK8s solution is equivalent to a Kubernetes Small Installation since their hardware requirements are the same. To make things more consistent, could we use the same kind of categories for both MicroK8s and Kubernetes? For example, something like this:
  * MicroK8s Small Installation (drop the "minimum requirements")
  * Kubernetes Small Installation
  * Kubernetes Medium Installation
  * Kubernetes Big Installation

  or

  * MicroK8s Minimum
  * MicroK8s Recommended
  * Kubernetes Minimum
  * Kubernetes Comfortable
  * Kubernetes Optimal

Reply:
> Depending on the type / size of installation, the user must adjust the values in the reference values-[production|microk8s].yaml files, correct? If so, this mapping should be made clearer below when we mention the reference .yaml files.

I don't think most customers need to change anything in the values-[production|microk8s].yaml files.

When the installation starts to get realllyyy big, maybe it is a good idea for customers to start increasing the number of replicas for each component. When that time comes, they need to edit (or create their one) the file and do the calculations taking into account the resources that are on those files. 

Example:

This is the codacy-api with 2 replicas:

```
codacy-api:
  replicaCount: 2
  resources:
    limits:
      cpu: 1000m
      memory: 2000Mi
    requests:
      cpu: 500m
      memory: 500Mi
```

If set it to one more replica, it will use more 500-1000 CPU and 500-2000 Memory in your cluster:

```
codacy-api:
  replicaCount: 3
  resources:
    limits:
      cpu: 1000m
      memory: 2000Mi
    requests:
      cpu: 500m
      memory: 500Mi
```

> What are the differences between a small / medium / big installation? Does it only have to do with the number of analysis running in parallel? Or are there any other factors that affect this, such as number of users simultaneously using the website or the number and size of the repositories?

The number of concurrent analysis is definitely the most impactful.

Check here the total requests and limits for 2 concurrent analysis (look at the `worker` and `tools`):

![image](https://user-images.githubusercontent.com/3610528/77421805-61cc0680-6dc4-11ea-95aa-0390cdaf217c.png)

10 concurrent analysis:

![image](https://user-images.githubusercontent.com/3610528/77421830-6db7c880-6dc4-11ea-9cbc-d45c12a5cf2c.png)


> MicroK8s Minimum
MicroK8s Recommended
Kubernetes Minimum
Kubernetes Comfortable
Kubernetes Optimal

I really like this

Reply:
OK, I think I was confusing the **infrastructure** provisioning (vCPUs, RAM memory, and storage) with the **cluster** provisioning (defined on the .yaml files).